CIScripts/Tests/TestConfigurationUtils.ps1

```diff
@@ -148,35 +148,50 @@ function Test-IsDockerDriverEnabled {
     return Test-IsProcessRunning -Session $Session -ProcessName "contrail-windows-docker"
 }
 
-function Enable-VRouterAgent {
-    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
-           [Parameter(Mandatory = $true)] [string] $ConfigFilePath)
-
-    Write-Host "Enabling Agent"
+function Enable-AgentService {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
 
     Invoke-Command -Session $Session -ScriptBlock {
-        $ConfigFilePath = $Using:ConfigFilePath
+        Write-Host "Starting Agent"
+        Start-Service ContrailAgent | Out-Null
+    }
+}
 
-        Start-Job -ScriptBlock {
-            Param ($ConfigFilePath)
+function Disable-AgentService {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
 
-            & "C:\Program Files\Juniper Networks\Agent\contrail-vrouter-agent.exe" --config_file $ConfigFilePath
-        } -ArgumentList $ConfigFilePath | Out-Null
+    Invoke-Command -Session $Session -ScriptBlock {
+        Write-Host "Stopping Agent"
+        Stop-Service ContrailAgent -ErrorAction SilentlyContinue | Out-Null
     }
 }
 
-function Disable-VRouterAgent {
+function Assert-IsAgentServiceEnabled {
     Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
 
-    Write-Host "Disabling Agent"
-
-    Stop-ProcessIfExists -Session $Session -ProcessName "contrail-vrouter-agent"
+    $Service = Invoke-Command -Session $Session -ScriptBlock {
+        return $(Get-Service "ContrailAgent" -ErrorAction SilentlyContinue)
+    }
+    if (!$Service) {
+        throw "Agent service is not registered. EXPECTED: Agent service registered"
+    }
+    if ($Service.Status -eq "Stopped") {
```

Comment:
Wouldn't it be better if we check "not equal Started". I am asking since there are some states like "Paused" or "Stop pending", which rather indicates that service is not enabled.